components/api/src/main/java/com/hotels/styx/api/MessageBodyConsumption.java

```diff
@@ -0,0 +1,38 @@
+/*
+  Copyright (C) 2013-2019 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.api;
+
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Mono;
+
+final class MessageBodyConsumption {
+    private MessageBodyConsumption() {
+    }
+
+    static <T extends LiveHttpMessage> Eventual<T> consume(T message, Class<T> type) {
```

Comment:
Please avoid overly clever code reuse. Follow the [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) instead.

There isn't any need to pass in class parameters etc because the message body stream is the same for both request and response types.